### PR TITLE
feat: adding a check for the use of conventional commits in pull request titles

### PR DESCRIPTION
**Pull request by OllyRich1**
This implements a GitHub action in which upon creating a PR the title checks to ensure the title adheres to the conventional commit standards. 

This ensures that all pull requests accurately describe the changes they have made

### CODE REVIEW

**Comment by Mahifernando** [comment]
Please provide appropriate PR description

File: .github/workflows/cicd-1-pull-request.yaml [hunk 1]
```diff
@@ -120,3 +120,11 @@ jobs:
       terraform_version: "${{ needs.metadata.outputs.terraform_version }}"
       version: "${{ needs.metadata.outputs.version }}"
     secrets: inherit
+  validate-title-stage:
```
Comment (Mahifernando) [inline .github/workflows/cicd-1-pull-request.yaml:123]: When PR title validation failed you need to write a comment to the PR for informing an Engineer what needs to be done for correcting their PR.

If your chosen action workflow doent support direct message writing upon failure make sure you  capture PR title validation output and using another strep for writing a custom message.

**Review by Mahifernando — CHANGES_REQUESTED** [review]

**Comment by Mahifernando** [comment]
Did you check PR title validation workflow on your fork?

For an example;

- Create a PR with conventional commits compliance title
- Create a PR WITHOUT conventional commits compliance title
- See if you get comments for wrong titles
- See what happens when you resolved PR title

**Comment by OllyRich1** [comment]
> Did you check PR title validation workflow on your fork?
> 
> For an example;
> 
>     * Create a PR with conventional commits compliance title
> 
>     * Create a PR WITHOUT conventional commits compliance title
> 
>     * See if you get comments for wrong titles
> 
>     * See what happens when you resolved PR title

yes, i have done all the above

**Review by Mahifernando — APPROVED** [review]

**Review by Mahifernando — APPROVED** [review]
Redy for merge 👍

**Pull request merged**